### ISSUE TITLE
Section 1.1 requires loading the entire app at once?

### ISSUE THREAD

**Issue by pes10k** [w3c/miniapp-packaging#60]
This issue is being filed as part of PING [wide review](https://github.com/w3cping/privacy-request/issues/99).

Section 1.1. seems to require the runtime load all mini-app resources at once, instead of loading them as needed.

> After retrieving the [MiniApp ZIP container](https://www.w3.org/TR/2022/WD-miniapp-packaging-20220729/#dfn-container), the [MiniApp user agent](https://www.w3.org/TR/2022/WD-miniapp-packaging-20220729/#dfn-useragent) loads the static page templates, stylesheets, JavaScript files, and other resources into the cache, following a [dereference and processing algorithm](https://www.w3.org/TR/2022/WD-miniapp-packaging-20220729/#sec-miniapp-processing). These [MiniApp resources](https://www.w3.org/TR/2022/WD-miniapp-packaging-20220729/#dfn-resource) will remain available in the cache until the next update, avoiding unnecessary network fetches.

This seems like it would be a very expensive choice, especially if the miniapp contains a lot of code or subpages that the user might never use.  Is this needed?  Are runtimes prohibited from loading resources on demand / lazing loading? If so, it might be worth reconsidering, since this seems an expensive option.  If not, i suggest clarifying Section 1.1.

This is not a blocking privacy issue, but I'm mentioning it since it came up during the review

**Comment by pes10k** [comment]
The above is especially surprising since this text would seem to require runtimes to load language files for languages other than the users current language.  I expect thats not intended, but i suggest tightening up the text here

**Comment by xfq** [comment]
Indeed, the current model requires loading the entire app at once, and we should do some work to reduce the resources loaded (like https://github.com/w3c/miniapp-packaging/issues/42 and https://github.com/w3c/miniapp-manifest/issues/19 ).

**Comment by pes10k** [comment]
@xfq , ah thats vert surprising.  The app should load all language files into memory, even for languages the user doesn't (and likely never will) use?  That seems like a very memory-expensive implementation choice!
